pkg/webhook/parser.go

```diff
@@ -238,7 +248,10 @@ func (c Config) sideEffects() *admissionreg.SideEffectClass {
 	case strings.ToLower(string(admissionreg.SideEffectClassSome)):
```

Comment:
In `admissionregistration/v1`, the sideEffects can only be `None` and `NoneOnDryRun`.
In the current implementation, users can still specify the sideEffects as `Some` or `Unknown` when genreate `v1` webhook configurations, but they will be rejectd by kubernetes api server.
Is that sufficient to handle it like this? I would like not to cut the above switch branches since we still need `Some` and `Unkown` for v1beta1. We can remove that when we drop the support for v1beta1.

Reply:
I can also add a check when generate `v1` version configurations, if it is set to `Some` or `Unknown` we default it to `None`
WDYT? @alvaroaleman @vincepri

Reply:
Can we just error out if its set to `Some` or `Unknown` rather than silently overwriting it? Users might not be aware that that got changed

Reply:
@alvaroaleman How about we didn't do anything for that? The reasons are as following:
1. `Some` and `Unknown` are still valid values for `SideEffects` for version `v1`, they can not be set directly when you create them. But if the user creates a webhookconfiguration with version `v1beta1` and with `Some` or `Unknown`, and when it is converted to version `v1` by apiserver, the Values are still valid. https://github.com/kubernetes/api/blob/master/admissionregistration/v1/types.go#L277
2. IMHO, we don't need even to overwrite it or do anything about it. If the user tries to apply the configuration to Kubernetes, it will be rejected, and then the user knows it should be `None` or `NoneOnDryRun`. and then they can update the marker and regenerate.  

WDYT?

Reply:
sgtm